### ISSUE TITLE
BUG: can't use annotate_timestamp with code units

### ISSUE THREAD

**Issue by vdbma** [yt-project/yt#3869]
### Bug report

**Bug summary**

When loading a data set using code units, `p.annotate_timestamp()` fails.

**Code for reproduction**

```python
import yt
from yt.testing import fake_random_ds

ds = fake_random_ds(64, unit_system="code")
p = yt.SlicePlot(ds,"x",("gas", "density"))
p.annotate_timestamp()
p.save("timestamp.png")
```

**Actual outcome**

```
$ python timestamp.py
yt : [INFO     ] 2022-03-29 16:08:23,834 Parameters: current_time              = 0.0
yt : [INFO     ] 2022-03-29 16:08:23,834 Parameters: domain_dimensions         = [64 64 64]
yt : [INFO     ] 2022-03-29 16:08:23,834 Parameters: domain_left_edge          = [0. 0. 0.]
yt : [INFO     ] 2022-03-29 16:08:23,835 Parameters: domain_right_edge         = [1. 1. 1.]
yt : [INFO     ] 2022-03-29 16:08:23,835 Parameters: cosmological_simulation   = 0
yt : [INFO     ] 2022-03-29 16:08:24,111 xlim = 0.000000 1.000000
yt : [INFO     ] 2022-03-29 16:08:24,111 ylim = 0.000000 1.000000
yt : [INFO     ] 2022-03-29 16:08:24,111 xlim = 0.000000 1.000000
yt : [INFO     ] 2022-03-29 16:08:24,111 ylim = 0.000000 1.000000
yt : [INFO     ] 2022-03-29 16:08:24,112 Making a fixed resolution buffer of (('gas', 'density')) 800 by 800
Traceback (most recent call last):
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_modifications.py", line 2530, in __call__
    un = self.time_unit.latex_representation()
AttributeError: 'Symbol' object has no attribute 'latex_representation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_window.py", line 1399, in run_callbacks
    callback(cbw)
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_modifications.py", line 41, in _check_geometry
    return func(self, plot)
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_modifications.py", line 2534, in __call__
    raise RuntimeError(
RuntimeError: The time unit str repr didn't match expectations, something is wrong.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "~/Documents/IPAG/ReposDivers/bugs_yt/timestamp.py", line 7, in <module>
    p.save("timestamp.png")
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_container.py", line 109, in newfunc
    args[0]._setup_plots()
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_window.py", line 1265, in _setup_plots
    self.run_callbacks()
  File "~/.pyenv/versions/yt-timestamp/lib/python3.9/site-packages/yt/visualization/plot_window.py", line 1403, in run_callbacks
    raise YTPlotCallbackError(callback._type_name) from e
yt.utilities.exceptions.YTPlotCallbackError: annotate_timestamp callback failed

```

**Version Information**

  * Operating System: Debian 5.10.106-1 x86_64 GNU/Linux
  * Python Version: 3.9.9
  * yt version: 4.0.2 (installed through `pip install yt`) and 4.1.dev0 (through `pip install -e .` in cloned repo)

**Comment by neutrinoceros** [comment]
Thanks Marc, really appreciate the report !
So unfortunately this seems reminiscent of a bunch of problems we've had in the past with "code" unit system. Here's a summary of what I could find back:
- #2353 (closed by #2354)
- #2432 (closed by #2435)
- other PR without an issue: #2728
- akin to #2913 (closed by #3618)

**Comment by neutrinoceros** [comment]
This bisects to fda512b6701d6590a5c8a83c821b6f4e0e7aa2a3 from #3618
